extensions/interactions/TextInput/TextInput.js

```diff
@@ -81,3 +81,21 @@ oppia.directive('oppiaShortResponseTextInput', [
     };
   }
 ]);
+
+oppia.factory('textInputRulesService', [function() {
+  return {
+    Equals: function(answer, inputs) {
+      return answer.toLowerCase() == inputs.x.toLowerCase();
+    },
+    CaseSensitiveEquals: function(answer, inputs) {
+      return answer == inputs.x;
+    },
+    StartsWith: function(answer, inputs) {
+      return answer.indexOf(inputs.x, 0) == 0;
+    },
+    Contains: function(answer, inputs) {
+      return answer.toLowerCase().indexOf(inputs.x.toLowerCase(), 0) != -1;
+    }
+
```

Comment:
Good point -- this was one of the first rules to be written, and I don't think it's quite consistent.

I suggest converting both x and the subject to lowercase, and only then doing the misspelling check. This does change the behaviour, but I think that's appropriate. (What do you think?)